### DISCUSSION TITLE
How to terminate stream properly?

### DISCUSSION THREAD

**Discussion by gohkokhan** [tweepy/tweepy#1657] [Questions] [answered]
I am building a Twitter monitoring telegram bot. User can send a command to stop the streaming at any time.
I am using 'return False' to terminate the stream as below: 
```
class MyStreamListener(tweepy.StreamListener):
    def on_data(self, data):
        if _FINISH:
            return False
```
The problem I am facing is that the stream only gets terminated when new data is coming in. Sometimes, user stops the stream, updates the filtering rule, and starts the stream again. It causes two streams to run simultaneously.

**Comment by Harmon758** [comment, marked as answer]
This sounds like #773, which has been resolved with https://github.com/tweepy/tweepy/commit/37e701e2193dcf617e7235d3b004f790c5ff784e in the latest development version on the master branch, set to be released as part of v4.0.
